modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +963,284 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - 3 * nlanes; x += 3 * nlanes)
+        {
+
+            v_float32 a1 = v_cvt_f32(vx_load_expand(in + x)),
+                      a2 = v_cvt_f32(vx_load_expand(in + x + nlanes / 2)),
+                      a3 = v_cvt_f32(vx_load_expand(in + x + nlanes)),
+                      a4 = v_cvt_f32(vx_load_expand(in + x + 3 * nlanes / 2)),
+                      a5 = v_cvt_f32(vx_load_expand(in + x + 2 * nlanes)),
+                      a6 = v_cvt_f32(vx_load_expand(in + x + 5 * nlanes / 2));
+
+            absdiffc_short_store_c3(&out[x], v_round(v_absdiff(a1, s1)),
+                                             v_round(v_absdiff(a2, s2)),
+                                             v_round(v_absdiff(a3, s3)),
+                                             v_round(v_absdiff(a4, s1)),
+                                             v_round(v_absdiff(a5, s2)),
+                                             v_round(v_absdiff(a6, s3)));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - 3 * nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<>
+CV_ALWAYS_INLINE int absdiffc_simd_c3_impl<uchar>(const uchar in[], uchar out[],
+                                                  const v_float32& s1, const v_float32& s2,
+                                                  const v_float32& s3, const int length)
+{
+    constexpr int nlanes = static_cast<int>(v_uint8::nlanes);
+
+    if (length < 3 * nlanes)
+        return 0;
+
+    int x = 0;
+    v_float32 vectors[12];
+    for (;;)
+    {
+        for (; x <= length - 3 * nlanes; x += 3 * nlanes)
+        {
+            for (int i = 0; i < 12; ++i)
+            {
+                vectors[i] = v_cvt_f32(vx_load_expand_q(reinterpret_cast<const uchar*>(in + x + i * nlanes / 4)));
+            }
+
+            vx_store(reinterpret_cast<uchar*>(&out[x]), v_pack_u(v_pack(v_round(v_absdiff(vectors[0], s1)),
+                                                                        v_round(v_absdiff(vectors[1], s2))),
+                                                                 v_pack(v_round(v_absdiff(vectors[2], s3)),
+                                                                        v_round(v_absdiff(vectors[3], s1)))));
+
+            vx_store(reinterpret_cast<uchar*>(&out[x + nlanes]), v_pack_u(v_pack(v_round(v_absdiff(vectors[4], s2)),
+                                                                                 v_round(v_absdiff(vectors[5], s3))),
+                                                                          v_pack(v_round(v_absdiff(vectors[6], s1)),
+                                                                                 v_round(v_absdiff(vectors[7], s2)))));
+
+            vx_store(reinterpret_cast<uchar*>(&out[x + 2 * nlanes]), v_pack_u(v_pack(v_round(v_absdiff(vectors[8], s3)),
```

Comment:
Do you really need to `reinterpret_cast` all of these? They are already `uchars` as it is `<uchar>` specialization

Reply:
Checked - compiles without casts

Reply:
Done